lib/stdlib/src/erl_expand_records.erl

```diff
@@ -68,16 +71,48 @@ init_calltype_imports([_|T], Ctype) ->
     init_calltype_imports(T, Ctype);
 init_calltype_imports([], Ctype) -> Ctype.
 
-forms([{attribute,_,record,{Name,Defs}}=Attr | Fs], St0) ->
+init_typetype(Forms) ->
+    init_typetype_imports(Forms, #{}).
+
+init_typetype_imports([{attribute,_,import_type,{Mod,Fs}}|T], Ttype0) ->
+    true = is_atom(Mod),
+    Ttype = foldl(fun(TA, Acc) -> Acc#{TA=>{imported,Mod}} end, Ttype0, Fs),
+    init_typetype_imports(T, Ttype);
+init_typetype_imports([_|T], Ttype) ->
+    init_typetype_imports(T, Ttype);
+init_typetype_imports([], Ttype) -> Ttype.
```

Comment:
Would it make sense if this is re-written in a foldl/map style? My reasoning is that folds/maps need to traverse the whole structure, so it is clear by its pattern what it does, and avoids dealing with the empty case pattern. Recursion may also be used to stop early a computation, which is not the case here.

```erlang
init_type(Forms) ->
    foldl(fun ({attribute,_,import_type,{Mod,Fs}}, Ttype0) when is_atom(Mod) ->
                  foldl(fun(TA, Acc) -> Acc#{TA=>{imported,Mod}} end, Ttype0, Fs);
              (_, Ttype0) ->  Ttype0
          end, #{}, Forms).
```
An alternative way which is even simpler, although not as efficient due to the `dict:from_list` follows:
```erlang
init_type(Forms) ->
    Imports = [{TA, {imported, Mod}}|| {attribute,_,import_type,{Mod,Fs}} <- Forms, is_atom(Mod), TA <- Fs],
    dict:from_list(Imports).
```

This is simply a brainstorming thought